lib/jekyll/static_file.rb

```diff
@@ -95,7 +95,7 @@ def write?
     #
     # Returns false if the file was not modified since last time (no-op).
     def write(dest)
-      dest_path = destination(dest)
+      dest_path = destination(dest).squeeze(".")
```

Comment:
Can you explain what this is doing and why it's needed in addition to the above?

Reply:
`String#squeeze` reduces multiple occurrences of a substring into a single occurrence.
```ruby
"my-cool-avatar...png".squeeze(".")  # => "my-cool-avatar.png"
```

> why it's needed in addition to the above?

The "above" sanitization is in a different class of code..
The "above" regex sanitization was in `Jekyll::Page` and `Jekyll::Document` where `basename` would determine the generated `url`

"This" is in `Jekyll::StaticFile`, where the `destination url` is *generally only* determined by the static-file's `path` and `relative_path`..
https://github.com/jekyll/jekyll/blob/fd6e01fd6a33365deaf19832f25f1b77e1d93697/lib/jekyll/static_file.rb#L133-L142

https://github.com/jekyll/jekyll/blob/fd6e01fd6a33365deaf19832f25f1b77e1d93697/lib/jekyll/static_file.rb#L119-L128